SwiftReflector/MarshalEngineCSafeSwiftToCSharp.cs

```diff
@@ -272,15 +272,17 @@ public MarshalEngineCSafeSwiftToCSharp (CSUsingPackages use, List<string> identi
 					invoker = new CSIndexExpression (registryCall, false, callParams.ToArray ());
 				}
 			} else {
-				if (thisIsInterface) {
+				if (thisIsInterface && !hasAssociatedTypes) {
+					var thisTypeName = hasAssociatedTypes ? csProxyName : thisType.ToString ();
 					invoker = new CSFunctionCall (
-						$"SwiftObjectRegistry.Registry.InterfaceForExistentialContainer<{thisType.ToString ()}> (self).{methodName}",
+						$"SwiftObjectRegistry.Registry.InterfaceForExistentialContainer<{thisTypeName}> (self).{methodName}",
 						false, callParams.ToArray ());
 
 				} else {
+					var thisTypeName = hasAssociatedTypes ? csProxyName : thisType.ToString ();
```

Comment:
This looks identical to the previous clause. Why not move it outside the `if` condition ?